lib/Backends/CPU/CMakeLists.txt

```diff
@@ -13,52 +13,58 @@ set(CMAKE_LLIR_CREATE_SHARED_LIBRARY "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
 set(CMAKE_LLIR_CREATE_SHARED_MODULE "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
 set(CMAKE_LLIR_CREATE_STATIC_LIBRARY "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
 
-add_library(CPURuntime
-              libjit/libjit.cpp
-              libjit/libjit_conv.cpp
-              libjit/libjit_matmul.cpp)
+set(CPURunttimeCompileOptions 
+      -ffast-math
+      -fno-finite-math-only
+      -g0
+      -emit-llvm 
+      -O0)
+
+set(CPURuntime_val CPURuntime)
+  
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${CPURuntime_val})
+file(GLOB_RECURSE libjit_SOURCES LIST_DIRECTORIES false libjit/*.cpp)
 
-# NOTE(abdulras) explicitly override the compiler invocations with a custom
-# rule.  The trailing `#` is the comment leader intended to nullify the compile
-# commands.  Doing this allows us to override the compiler (driver) used for
-# building the runtime which requires clang (to emit LLVM IR for the LTO'ed AOT
-# JIT runtime).  Use a custom linker language with the rule specified above to
-# invoke our custom linker to merge the bitcode files.
-set_target_properties(CPURuntime
-                      PROPERTIES
-                        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
-                        CXX_STANDARD 11
-                        CXX_STANDARD_REQUIRED YES
-                        LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
-                        LINKER_LANGUAGE LLIR
-                        OUTPUT_NAME libjit.bc
-                        POSITION_INDEPENDENT_CODE YES
-                        PREFIX ""
-                        SUFFIX ""
-                        RULE_LAUNCH_COMPILE "${CLANG_BIN} <DEFINES> <INCLUDES> <FLAGS> -o <OBJECT> -c <SOURCE> #")
+set(CPURuntime_OBJS)
 
-# Due to the usage for NaN in the runtime, '-fno-finite-math-only' is needed
-# to disable '-ffinite-math-only' activated by '-ffast-math', while benefiting
-# from math optimizations.
-target_compile_options(CPURuntime
-                       PRIVATE
-                         -ffast-math
-                         -fno-finite-math-only
-                         -g0
-                         -emit-llvm
-                         -O0)
+foreach(libjit_src_file ${libjit_SOURCES})
+  MATH(EXPR INCR_VAR "${INCR_VAR}+1")    
+  set(libjit_obj_file libjit_obj_file_${INCR_VAR}.o)
+  set(libjit_obj_file_path ${CMAKE_CURRENT_BINARY_DIR}/${CPURuntime_val}/${libjit_obj_file})
+
+  add_custom_command(
+    OUTPUT ${libjit_obj_file_path}
+    COMMAND ${CLANG_BIN} -c ${libjit_src_file} ${CPURunttimeCompileOptions} -o ${libjit_obj_file_path}
+    DEPENDS ${libjit_src_file}
+    WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
+  )
+  list(APPEND CPURuntime_OBJS ${libjit_obj_file_path})
+endforeach()
```

Comment:
You should be able to use an object library here for this.  If you do that, you do not need to do the object file name computation and can use the `TARGET_OBJECTS` generator.

Reply:
You can still add the sources (objects) expanding the target objects.

Reply:
Downside of above approach is that it will always rebuild object files.